app.go

```diff
@@ -614,9 +615,14 @@ func (a *Application) processTx(txBytes []byte, isCheckTx bool) (TxHandlerResult
 
 	r, err := a.TxHandler.ProcessTx(state, txBytes, isCheckTx)
 	if err != nil {
-		storeTx.Rollback()
-		// TODO: save receipt & hash of failed EVM tx to node-local persistent cache (not app state)
-		receiptHandler.DiscardCurrentReceipt()
+		evmRevertedError := strings.Contains(err.Error(), "evm: execution reverted")
```

Comment:
Nah, this is too hacky to be relied on. Need a better way.

Reply:
The error returned is https://github.com/ethereum/go-ethereum/blob/aa6005b469fdd1aa7a95f501ce87908011f43159/core/vm/instructions.go#L35. We can only compare string. Maybe to make it better I can do
```
err.Error() == "evm: execution reverted"
```

Reply:
Can't rely on error strings matching, what if it errors out for a different reason?